### PR TITLE
Upgrade pyopenssl version to fix OpenSSL issue; slightly refactor SSL cert generation and add tests

### PR DESCRIPTION
**Pull request by whummer**
* upgrade pyopenssl version to fix recent OpenSSL issue
* slightly refactor SSL cert generation and add tests
* minor fix for `on_stop` handler with Tailer

### CODE REVIEW

File: tests/unit/utils/test_common.py [hunk 1]
```diff
@@ -7,11 +8,18 @@
...
+    PEM_KEY_START_REGEX,
     FileListener,
+    generate_ssl_cert,
     is_none_or_empty,
+    load_file,
+    new_tmp_file,
     poll_condition,
     run,
-    synchronized,
+    synchronized, rm_rf,
```
Comment (thrau) [inline tests/unit/utils/test_common.py:22]: i think this breaks the linter

**Review by thrau — APPROVED** [review]
LGTM! (once the tests run through)

i don't know what the implications are with pyopenssl with respect to the dependency tree. hopefully the upgrade will be smooth!

File: localstack/utils/common.py [hunk 2]
```diff
@@ -1689,17 +1695,18 @@ def all_exist(*files):
     def store_cert_key_files(base_filename):
         key_file_name = "%s.key" % base_filename
         cert_file_name = "%s.crt" % base_filename
-        # TODO: Cleaner code to load the cert dinamically
+        # TODO: Cleaner code to load the cert dynamically
         # extract key and cert from target_file and store into separate files
         content = load_file(target_file)
-        key_start = re.search(r"-----BEGIN(.*)PRIVATE KEY-----", content)
+        print(content)
```
Comment (dfangl) [inline localstack/utils/common.py:1701]: Any reason for this print here, or is this a leftover?

**Review by dfangl — APPROVED** [review]
LGTM :+1:

File: localstack/utils/common.py (hunk 2 above)
Comment (whummer) [reply localstack/utils/common.py]: Thanks! 👍

**Pull request merged**